### ISSUE TITLE
Why is kid mandatory for cnf claims?

### ISSUE THREAD

**Issue by paulbastian** [vcstuff/draft-terbu-sd-jwt-vc#97]
The spec says:

> cnf
REQUIRED when Cryptographic Holder Binding is to be supported. Contains the confirmation method as defined in [[RFC7800](https://vcstuff.github.io/draft-terbu-sd-jwt-vc/draft-terbu-sd-jwt-vc.html#RFC7800)]. It SHOULD contain a JWK as defined in Section 3.2 of [[RFC7800](https://vcstuff.github.io/draft-terbu-sd-jwt-vc/draft-terbu-sd-jwt-vc.html#RFC7800)] and in this case, the kid (Key ID) member MUST be present in the JWK. For Cryptographic Holder Binding, the Holder Binding JWT in the Combined Format for Presentation MUST be signed by the key identified in this claim.

I don't see why `kid` is a MUST.
Moreover, the example in 4.3 does not include the `kid`, so either the text or the exampel is wrong.

**Comment by bc-pi** [comment]
Yeah, the `kid` member in the cnf JWK shouldn't be a MUST. It probably shouldn't be mentioned at all.

**Comment by bc-pi** [comment]
I don't follow what ambiguity there might be? 

A `cnf` claim with the `jwk` member (having only the required parameters) unambiguously has the whole public key that corresponds to the private key that signs the Key Binding JWT. That's it. And that's sufficient. That's what all the examples in SD-JWT have (like [this one](https://www.ietf.org/archive/id/draft-ietf-oauth-selective-disclosure-jwt-04.html#section-5.2-4)) by the way and as @paulbastian pointed out, the [example in SD-JWT VC](https://datatracker.ietf.org/doc/html/draft-terbu-sd-jwt-vc-02#section-4.3-4) is the same (a `cnf` with a `jwk` having only the required parameters). 

The text, "and in this case, the kid (Key ID) member MUST be present in the JWK" can just be removed from the spec.

**Comment by paulbastian** [comment]
@awoie I don't understand your concerns. Can you provide examples?

**Comment by paulbastian** [comment]
@awoie I don't understand your concerns, can you provide examples?

**Comment by awoie** [comment]
I think we are good here and we should remove the `kid` requirement as @bc-pi suggested. 

Especially because we have the following requirement...

> Holder Binding, the Holder Binding JWT in the Combined Format for Presentation MUST be signed by the key identified in this claim.

My concern was that technically the following is possible as well:

```
"cnf": {
  "kid": "some-kid-1",
  "jwk": {
    "kid": "another-kid-2",
    ....
  }
}
```

That could lead to ambiguities and the verifier might need to test both confirmation method to verify the key binding JWT. If this is not possible by another requirement in either RFC7800 or the SD-JWT draft, then I might have overlooked that.

**Comment by awoie** [comment]
Needs PR to remove requirements on `kid` in `cnf`.

**Comment by bc-pi** [comment]
PR #141 removes requirements to have kid in jwk in cnf

@awoie I think you are right that RFC7800 doesn't strictly prohibit something like that but it seems generally understood and I've never encountered a cnf claim with more than one member in it

**Comment by Sakurann** [comment]
I don't think RFC7800 allows for multiple JWKs in one `cnf` claim: https://www.rfc-editor.org/rfc/rfc7800#section-3.2

and I think it indirectly prohibits the following example Oliver is concerned with because section 3.2 limits cnf claim with only jwk member and only mentions that that jwk member may contain a `kid`:
```
"cnf": {
  "kid": "some-kid-1",
  "jwk": {
    "kid": "another-kid-2",
    ....
  }
}
```

so all good removing kid requirement with jwk member in cnf claim
